### DISCUSSION TITLE
[BUG] Set-PnPPageWebPart: Cannot set images for image / image gallery web part

### DISCUSSION THREAD

**Discussion by robi26** [pnp/powershell#1071] [Q&A]
Hi
I'm trying to add and configure an image / image gallery web part to my pages using the following code:
```powershell
$imgWebPart = Add-PnPPageWebPart -Page $page -DefaultWebPartType Image -Section 1 -Column 1
$json = $imgWebPart.PropertiesJson | ConvertFrom-Json
$json.siteId = $site.Id
$json.webId = $web.Id
$json.listId = $imageList.Id
$json.uniqueId = $images[0].id
Set-PnPPageWebPart -Page $page -Identity $imgWebPart.InstanceId.Guid -PropertiesJson $(ConvertTo-Json $json)
```
The $json looks as follows: 
```json
{
  "imageSourceType": 0,
  "imageSource": "",
  "captionText": "",
  "altText": "",
  "linkUrl": "",
  "overlayText": "",
  "fileName": "",
  "siteId": "139f5750-2d56-4783-8568-fad2668a1dd3",
  "webId": "1f25310d-b433-4153-b50f-4e9f4c19aaca",
  "listId": "3d17631e-2cdf-4584-a572-613cc8c5ccc5",
  "uniqueId": "9658870c-6a44-4f26-9bfe-33725bb5915a",
  "imgWidth": "",
  "imgHeight": ""
}
```

The web part is added to the page (shown when in edit-mode) but no image is displayed. Same when trying to add an image gallery.
Even getting the PropertiesJson from a manually configured web part and re-setting it with set-pnppagewebpart does not work.

I also tried to set the ``imageSource`` property to the URL of the image. Doesn't help.

### Expected behavior 
Images are displayed.

Thanks for any help / ideas...

**Comment by jessierobbins** [comment]
I'm having the exact same issue but also populating imgWidth, imgHeight, imageSource, altText, and imageSourceType is set to 2.  

Images will not display until the page is put into edit mode.  Even if I put the page into maintenance mode, it does not recognize the image web parts until the page has been edited.

**Reply by JoeAlanis** [reply]
I am having the same issue. The image wont display until I open the page in Edit mode in the Browser and re-publish
